medusa/server/api/v2/config.py

```diff
@@ -596,72 +599,73 @@ def data_main():
+
+    @classmethod
+    def data_consts(cls):
+        """Constant values - values that will never change during runtime."""
+        if cls._generated_data_consts:
+            return cls._generated_data_consts
+
         section_data = {}
 
-        section_data['values'] = {}
-        section_data['values']['na'] = common.Quality.NA
-        section_data['values']['unknown'] = common.Quality.UNKNOWN
-        section_data['values']['sdtv'] = common.Quality.SDTV
-        section_data['values']['sddvd'] = common.Quality.SDDVD
-        section_data['values']['hdtv'] = common.Quality.HDTV
-        section_data['values']['rawhdtv'] = common.Quality.RAWHDTV
-        section_data['values']['fullhdtv'] = common.Quality.FULLHDTV
-        section_data['values']['hdwebdl'] = common.Quality.HDWEBDL
-        section_data['values']['fullhdwebdl'] = common.Quality.FULLHDWEBDL
-        section_data['values']['hdbluray'] = common.Quality.HDBLURAY
-        section_data['values']['fullhdbluray'] = common.Quality.FULLHDBLURAY
-        section_data['values']['uhd4ktv'] = common.Quality.UHD_4K_TV
-        section_data['values']['uhd4kwebdl'] = common.Quality.UHD_4K_WEBDL
-        section_data['values']['uhd4kbluray'] = common.Quality.UHD_4K_BLURAY
-        section_data['values']['uhd8ktv'] = common.Quality.UHD_8K_TV
-        section_data['values']['uhd8kwebdl'] = common.Quality.UHD_8K_WEBDL
-        section_data['values']['uhd8kbluray'] = common.Quality.UHD_8K_BLURAY
-
-        section_data['anySets'] = {}
-        section_data['anySets']['anyhdtv'] = common.Quality.ANYHDTV
-        section_data['anySets']['anywebdl'] = common.Quality.ANYWEBDL
-        section_data['anySets']['anybluray'] = common.Quality.ANYBLURAY
-
-        section_data['presets'] = {}
-        section_data['presets']['any'] = common.ANY
-        section_data['presets']['sd'] = common.SD
-        section_data['presets']['hd'] = common.HD
-        section_data['presets']['hd720p'] = common.HD720p
-        section_data['presets']['hd1080p'] = common.HD1080p
-        section_data['presets']['uhd'] = common.UHD
-        section_data['presets']['uhd4k'] = common.UHD_4K
-        section_data['presets']['uhd8k'] = common.UHD_8K
-
-        section_data['strings'] = {}
-        section_data['strings']['values'] = common.Quality.qualityStrings
-        section_data['strings']['anySets'] = common.Quality.combinedQualityStrings
-        section_data['strings']['presets'] = common.qualityPresetStrings
-        section_data['strings']['cssClass'] = common.Quality.cssClassStrings
+        section_data['qualities'] = {}
+
+        # Get key names from `common.Quality` class (for the quality values and any-sets)
+        quality_key_map = {
+            value: text_type(key).lower().replace('_', '')
+            for (key, value)
+            in iteritems(common.Quality.__dict__)
+            if isinstance(value, int) and not key.startswith('__')
+        }
+
+        def make_quality(value, name, key=None):
```

Comment:
Same as above.

Reply:
I don't see the point in doing that.
1. Only used in this context, for the API data, probably won't have any other uses for it.
2. The `data_consts` function only generates the data once per Medusa instance, so it's not performance.

But prove me otherwise and I'll move it..

Reply:
In that case you don't need to move it. It's just really hard to keep an overview of what is already there this way. I mean, maybe in a few moths somebody else will need to do exactly the same and will push similar code. If we keep it all in one place it will be easier to avoid duplicates. Same goes with the `quality_key_map`. Also if one day we want to refactor the terrible Quality class, it might be easier.

Reply:
Thank you for that. I'll keep that in mind. One more thing about this specific function is that it's simply a helper function.
You know, like, when you feel like you write the same code for more than 2 times, you want it to not be this way? 😄 It just creates a dict. Makes it easier to read in the context of a list comprehension.